lib/chef/provisioning/docker_driver/docker_transport.rb

```diff
@@ -72,7 +72,29 @@ def read_file(path)
     end
 
     def write_file(path, content)
-      File.open(container_path(path), 'w') { |file| file.write(content) }
+      server = Socket.new(:INET, :STREAM, 0)
+      ip = Socket.ip_address_list.detect{|intf| intf.ipv4_private?}
+      ip = ip.ip_address if ip
+      server.bind(Addrinfo.tcp(ip, 0))
+      port = server.local_address.ip_port
+      server.close
+      server = TCPServer.new(port)
+      Thread.new do
+        client = server.accept
+        client.print content
+        client.flush
+        client.close
+      end
+      execute "mkdir -p #{File.dirname path}"
+      res = 1
+      while res != 0
+        if execute("which ruby").exitstatus == 0
```

Comment:
This isn't going to work. Most of the containers don't have nc within (try centos images for instance). Embedded ruby will only be available after chef omnibus is installed inside container. Installation process itself uses write file to copy omnibus archive to host.